src/psbtv2.ts

```diff
@@ -676,6 +690,63 @@ export class PsbtV2 {
     );
   }
 
+  /**
+   * Other Getters/Setters
+   */
+
+  get nLockTime () {
+    // From BIP0370: The nLockTime field of a transaction is determined by
+    // inspecting the PSBT_GLOBAL_FALLBACK_LOCKTIME and each input's
+    // PSBT_IN_REQUIRED_TIME_LOCKTIME and PSBT_IN_REQUIRED_HEIGHT_LOCKTIME
+    // fields.
+    //
+    // First collect total locks
+    const inputCount = this.PSBT_GLOBAL_INPUT_COUNT;
+    const heightLocks = this.PSBT_IN_REQUIRED_HEIGHT_LOCKTIME;
+    const timeLocks = this.PSBT_IN_REQUIRED_TIME_LOCKTIME;
+    let heights:number[] = [];
+    let times:number[] = [];
+    for (let i = 0; i < this.PSBT_GLOBAL_INPUT_COUNT; i++) {
+      if (heightLocks[i] !== null) {
+        heights.push(heightLocks[i] as number);
+      } 
+      
+      if (timeLocks[i] !== null) {
+        times.push(timeLocks[i] as number);
+      }
+    }
+
+    // From BIP0370: If none of the inputs have a PSBT_IN_REQUIRED_TIME_LOCKTIME
+    // and  PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, then PSBT_GLOBAL_FALLBACK_LOCKTIME
+    // must be used. If PSBT_GLOBAL_FALLBACK_LOCKTIME is not provided, then it
+    // is assumed to be 0.
+    if (heights.length === 0 && times.length === 0) {
+      return this.PSBT_GLOBAL_FALLBACK_LOCKTIME || 0;
+    }
+
+    // From BIP0370: If one or more inputs have a PSBT_IN_REQUIRED_TIME_LOCKTIME
+    // or PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, then the field chosen is the one
+    // which is supported by all of the inputs. This can be determined by
+    // looking at all of the inputs which specify a locktime in either of those
+    // fields, and choosing the field which is present in all of those inputs.
+    // Inputs not specifying a lock time field can take both types of lock
+    // times, as can those that specify both. The lock time chosen is then the
+    // maximum value of the chosen type of lock time.
```

Comment:
This is what you're saying you weren't sure about in your PR comment? I think the thinking goes that only the largest one will hold any "authority". Basically until that lock time is passed then it doesn't matter what the others say. I know there is some bad edge case where you can have an input that locks by block and one that locks by time and that they can be in conflict, but i don't think any of that is of concern here.

Reply:
actually might just be a complication within a single script so slightly different: https://medium.com/blockstream/dont-mix-your-timelocks-d9939b665094

Reply:
That isn't the line I was concerned with. Take this excerpt from the bip:
```
If one or more inputs have a PSBT_IN_REQUIRED_TIME_LOCKTIME
or PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, then the field chosen is the one
which is supported by all of the inputs. This can be determined by
looking at all of the inputs which specify a locktime in either of those
fields, and choosing the field which is present in all of those inputs.
Inputs not specifying a lock time field can take both types of lock
times, as can those that specify both. The lock time chosen is then the
maximum value of the chosen type of lock time.

If a PSBT has both types of locktimes possible because one or more inputs
specify both PSBT_IN_REQUIRED_TIME_LOCKTIME and
PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, then locktime determined by looking at
the PSBT_IN_REQUIRED_HEIGHT_LOCKTIME fields of the inputs must be chosen.
```

I tried only applying a locktime other than global fallback when all inputs have a locktime specified (as the bip seems to describe when it says ` If one or more inputs have a PSBT_IN_REQUIRED_TIME_LOCKTIME
or PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, then the field chosen is the one
which is supported by all of the inputs`), but one/some of the tests were failing. Based on the test vectors, perhaps what the the bip is trying to describe is to look at all inputs which have a locktime specified, then from those pick the one which is supported by all those inputs? I think I tried that and tests were still failing, but maybe I didn't do it correctly.

The way I've done it is to check and see if heights is supported by all and if so use that, otherwise check for the majority type.

Reply:
> but one/some of the tests were failing. Based on the test vectors, perhaps what the the bip is trying to describe is to look at all inputs which have a locktime specified, then from those pick the one which is supported by all those inputs?

Yeah that's how I read it, but agree it's very confusing.